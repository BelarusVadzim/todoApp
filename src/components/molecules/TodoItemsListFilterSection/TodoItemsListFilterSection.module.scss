@import "src/variables.scss";

.radioButton {
  display: flex !important;
  min-width: 1rem;
  align-items: center;
  justify-content: center;
  font-size: $todoItemsListMenu-font-size !important;
  margin: 1rem 0;
}
