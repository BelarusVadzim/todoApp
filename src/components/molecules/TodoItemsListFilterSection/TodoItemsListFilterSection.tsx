import React from 'react';
import { RadioButton } from 'components/atoms';
import style from './TodoItemsListFilterSection.module.scss';
import { useRadioGroup } from 'hooks';
import { PropsWithClassName } from 'types';
import { filterTypes } from 'constants/filterTypes';

const radioGroupName = 'radioGroup1';

const TodoItemsListFilterSection: React.FC<PropsWithClassName> = ({
  className = '',
}) => {
  const [createNewRadioButton] = useRadioGroup(radioGroupName);

  const allRadioButtonProps = createNewRadioButton(filterTypes.All, true);
  const activeRadioButtonProps = createNewRadioButton(filterTypes.Active);
  const completedRadioButtonProps = createNewRadioButton(filterTypes.Completed);

  return (
    <div className={className}>
      <RadioButton className={style.radioButton} {...allRadioButtonProps}>
        All
      </RadioButton>
      <RadioButton className={style.radioButton} {...activeRadioButtonProps}>
        Active
      </RadioButton>
      <RadioButton className={style.radioButton} {...completedRadioButtonProps}>
        Completed
      </RadioButton>
    </div>
  );
};

export default TodoItemsListFilterSection;
