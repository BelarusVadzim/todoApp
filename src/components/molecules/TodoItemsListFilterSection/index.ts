import TodoItemsListFilterSection from './TodoItemsListFilterSection';

export { TodoItemsListFilterSection };