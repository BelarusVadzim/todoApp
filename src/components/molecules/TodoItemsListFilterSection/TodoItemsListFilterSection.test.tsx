import { render } from '@testing-library/react';
import TodoItemsListFilterSection from './TodoItemsListFilterSection';

let createNewRadioButton = jest.fn();

jest.mock('./TodoItemsListFilterSection.module.scss', () => ({
  radioButton: 'radioButtonStyle',
}));

jest.mock('hooks', () => ({
  useRadioGroup: () => [createNewRadioButton],
}));

jest.mock('components/atoms', () => ({
  RadioButton: jest.fn(({ className, children }) => {
    return <div className={className}>{children}</div>;
  }),
}));

afterEach(() => {
  jest.clearAllMocks();
});

describe('<TodoItemsListFilterSection />', () => {
  it('should render properly', () => {
    const component = render(
      <TodoItemsListFilterSection className="mockClassName" />
    );

    expect(component.asFragment()).toMatchSnapshot();
  });

  it('when className not specified should render properly', () => {
    const component = render(<TodoItemsListFilterSection />);

    expect(component.asFragment()).toMatchSnapshot();
  });
});
