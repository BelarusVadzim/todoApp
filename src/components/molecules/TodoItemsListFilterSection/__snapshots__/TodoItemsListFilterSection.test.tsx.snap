// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoItemsListFilterSection /> should render properly 1`] = `
<DocumentFragment>
  <div
    class="mockClassName"
  >
    <div
      class="radioButton"
    >
      All
    </div>
    <div
      class="radioButton"
    >
      Active
    </div>
    <div
      class="radioButton"
    >
      Completed
    </div>
  </div>
</DocumentFragment>
`;

exports[`<TodoItemsListFilterSection /> when className not specified should render properly 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <div
      class="radioButton"
    >
      All
    </div>
    <div
      class="radioButton"
    >
      Active
    </div>
    <div
      class="radioButton"
    >
      Completed
    </div>
  </div>
</DocumentFragment>
`;
